File: library/src/main/java/com/jerrylee/lib/mvvm/BaseActivity.java
package com.jerrylee.lib.mvvm;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;

import butterknife.ButterKnife;
import butterknife.Unbinder;
import me.yokeyword.fragmentation.SupportActivity;

/**
 * Description:
 * Author: jerry Lee
 * Date: 2019/06/27 16:50
 */
public abstract class BaseActivity extends SupportActivity{
    private Unbinder unbinder;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResId());
        unbinder = ButterKnife.bind(this);
        initConfig(savedInstanceState);
        initUI(savedInstanceState);
        initData(savedInstanceState);
    }

    /**
     * 初始化UI
     */
    public abstract void initUI(@Nullable Bundle savedInstanceState);

    protected void initData(Bundle savedInstanceState){}

    protected void initConfig(Bundle savedInstanceState){}

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (unbinder != null){
            unbinder.unbind();
        }
    }

    protected abstract int getLayoutResId();
}

File: library/src/main/java/com/jerrylee/lib/mvvm/BaseActivity.kt
package com.jerrylee.lib.mvvm

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.jerrylee.lib.widget.stateview.core.LoadManager

/**
 * Description:
 * Author: jerry Lee
 * Date: 2019/06/27 16:50
 */
abstract class BaseActivity : AppCompatActivity() {

    protected var loadManager: LoadManager? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        beforeSetContentView(savedInstanceState)
        setContentView(getLayoutResId())
        val contentView:View? = findViewById(getContentId())
        contentView?.let {
            loadManager = LoadManager.Builder()
                    .setViewParams(findViewById(getContentId()))
                    .setListener{onStateRefresh()}
                    .build()
        }
        initConfig(savedInstanceState)
        initUI(savedInstanceState)
        initData(savedInstanceState)
    }

    /**
     * 初始化UI
     */
    abstract fun initUI(savedInstanceState: Bundle?)

    protected abstract fun getLayoutResId():Int

    /**
     * 获取需要显示loadingView的视图id
     */
    protected open fun getContentId():Int{
        return -1
    }

    protected open fun initData(savedInstanceState: Bundle?) {}

    protected open fun initConfig(savedInstanceState: Bundle?) {}

    protected open fun beforeSetContentView(savedInstanceState: Bundle?) {}

    protected open fun onStateRefresh(){}
}